### ISSUE TITLE
[Bug] Checkbox state does not follow its ngModel value

### ISSUE THREAD

**Issue by Etchelon** [primefaces/primeng#2162]
**I'm submitting a ...**  (check one with "x")
[ x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35

**Plunkr Case (Bug Reports)**
http://plnkr.co/edit/DGWj95FlcDFsfecaHucC?p=preview

**Current behavior**
Checking another checkbox should set all others to unchecked state and set the clicked one to checked state, programmatically, but does NOT uncheck the others even if their ngModel is set to false.

**Expected behavior**
Other checkboxes should be unchecked programmatically.

**Minimal reproduction of the problem with instructions**
See the plunkr, very very easy,

* **Angular version:** 2.0.X
2.4.8

* **PrimeNG version:** 2.0.X
2.0.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
All

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript >= v2

**Comment by external-olgierd** [comment]
I dont think its an issue, I think its a bug in your code in the plunker.

You bind the checkboxes to "checked" property of the your model items but when updating values you change the "isChecked" property instead so that change is never detected by the component.

I have forked your plunker and updated some code and it seems to work fine now - please have a look.

http://plnkr.co/edit/R5otr5DSB6dt99gqQStO?p=preview

**Comment by Etchelon** [comment]
You are right, I mistyped the code in the plunkr, but incidentally the outcome is exactly the bugged behavior I am experiencing. The bug is there, but I've found out that it could be related to Lodash, not primeng... so I'll close this issue, but fyi check this out:

This is working as intended. I loop the array with the Array.forEach function: http://plnkr.co/edit/DGWj95FlcDFsfecaHucC?p=preview
This is not working: if you click the checkboxes multiple times you'll experience the bug. I loop the array with _.each: http://plnkr.co/edit/FMHv213puJ387P5phO0k?p=preview
This is working: using Lodash but with a lambda with a body!! http://plnkr.co/edit/cr5TYBMUEKkANKXh024j?p=preview
